usd_qtpy/render_util/turntable.py

```diff
@@ -0,0 +1,374 @@
+                                            ) 
+    
+    subject_ref_xformable.AddTranslateOp(
+                          UsdGeom.XformOp.PrecisionDouble,"center_centroid")\
+                         .Set(subject_center_translate)
+    
+    # turn off the lights if GL
+    if renderer == "GL":
+        lights_prim = ttable_stage.GetPrimAtPath("/turntable_reference/scene/lights")
+        if lights_prim.IsValid():
+            lights_prim.GetAttribute("visibility").Set("invisible",0)
+
+    realstage_filename = R"./temp/turntable_assembly.usd"
+    realstage_filename = os.path.abspath(realstage_filename)
+
+    ttable_stage.Export(realstage_filename)
+
+    # frame range 1-100 in standard file
+    # get_file_timerange_as_string should be preferred, but it doesn't work atm.
+    frames_string = get_turntable_frames_string(length,frame_start,repeats)
+    render_path = os.path.join(export_path, "turntablefile_###.png")
+    render_path = os.path.abspath(render_path)
+
+    print("Rendering",frames_string,render_path)
+
+    realstage = Usd.Stage.Open(realstage_filename)
+
+    turntable_camera = next(playblast.iter_stage_cameras(realstage),None)
+    turntable_camera = UsdGeom.Camera(turntable_camera)
+    turntable_camera = framing_camera.camera_conform_sensor_to_aspect(
+        turntable_camera,
+        width,
+        height
+    )
+
+    playblast.render_playblast(realstage,
+                               render_path,
+                               frames=frames_string,
+                               width=1920,
+                               camera=turntable_camera,
+                               renderer=renderer,
+                               qt_report_instance=qt_report_instance)
+    
+    # explicitly free scene to make composite file available for deletion
+    del realstage
+
+    os.remove(subject_filename)
+    os.remove(realstage_filename)
+
+
+def file_is_zup(path: str) -> bool:
+    stage = Usd.Stage.CreateInMemory(path)
+    return framing_camera.get_stage_up(stage) == "Z"
+
+
+def get_file_timerange_as_string(path: str) -> str:
+    """
+    Attempt to get timerange from a USD file.
+
+    DOES NOT APPEAR TO WORK, EVEN WITH CORRECT METADATA.
+    """
+    stage = Usd.Stage.CreateInMemory(path)
+
+    if stage.HasAuthoredTimeCodeRange():
+        start = int(stage.GetStartTimeCode())
+        end = int(stage.GetEndTimeCode())
+    else:
+        print("No Timecode found")
+        start = 0
+        end = 100
+
+    return playblast.get_frames_string(start,end)
+
+
+def layer_from_layereditor(layer_editor:
+                           LayerTreeWidget) -> Union[Sdf.Layer, None]:
+    """
+    Get current selected layer in layer view, 
+    if none selected, return top of the stack.
+    """
+    
+    if index := layer_editor.view.selectedIndexes():
+        layertree_index = index[0]
+    else:
+        layertree_index = layer_editor.view.indexAt(QtCore.QPoint(0, 0))
```

Comment:
This doesn't feel safe as margins/padding or whatever might make the index not be at the 0, 0 point. I think you might need to go through the model instead getting the first row?

Reply:
Margins and padding will not affect the point where the index is at,
the is point is in "list space", meaning that it pertains to the model, not the location.

Even so, I'm not sure I want to keep this function in there. 
I thought I needed it, but I haven't used it at all because I stopped using the reparenting function.
instead, if I need a hierarchy to work a certain way I generate elements in order or, if I can't
I'll parent them. This saves me from having to pass in another parameter as well, which is always preferred.

Reply:
In that, if unused, case let's remove it.

Reply:
I'll remove it along with some other unused functions.
I had a different approach to rotating the subject at first, but the one I've landed on right now works 100% of the time